gobblin-service/src/main/java/org/apache/gobblin/service/modules/flow/MultiHopFlowCompiler.java

```diff
@@ -201,13 +202,19 @@ public void awaitHealthy() throws InterruptedException {
       for (FlowSpec datasetFlowSpec : flowSpecs) {
         for (DataNode destNode : destNodes) {
           long authStartTime = System.nanoTime();
-          boolean authorized = this.dataMovementAuthorizer.isMovementAuthorized(flowSpec, sourceNode, destNode);
-          Instrumented.updateTimer(dataAuthorizationTimer, System.nanoTime() - authStartTime, TimeUnit.NANOSECONDS);
-          if (!authorized) {
-            String message = String.format("Data movement is not authorized for flow: %s, source: %s, destination: %s",
-                flowSpec.getUri().toString(), source, destination);
-            log.error(message);
-            datasetFlowSpec.getCompilationErrors().add(message);
+          try {
```

Comment:
So for the NPE we experienced previously, it happened inside this block?

Reply:
Yes it happened in the DMASAuthorizer